0000-catalog.md

```diff
@@ -0,0 +1,152 @@
+
+Replication of a catalog is restricted to agents acting on behalf of an
+organization. The purpose of replication is to enable agents to duplicate a
+catalog in state with the intention of modifying the catalog for a different
+specification. It reduces the overhead needed to recreate a catalog only to
+add/modify few or many attributes of a catalog, or the products associated.
+
+In the case of **catalog operations** agents from the org they represent will
+need the operational permission to perform the desired operation on a Grid
+Catalog.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The primary object stored in state is **Catalog**, which consists of a
+catalog_id ([UUID](https://crates.io/crates/uuid)), an owner (org_id compatible
+w/Pike), an expiry_date (unix timestamp), an *optional* **Name**, and a list of
+property key/value pairs.  The properties available are defined by the Grid
+Property Schema Transaction Family and are restricted to the fields and rules of
+the GS1 Catalog schema (which will be defined at a later time). Transactions
+which are responsible for setting product state values must ensure that those
+properties conform with the requirements of the GS1 Catalog Property Schema (to
+be defined at a later time).
+
+``` 
+message Catalog { 
+    string catalog_id = 1; 
+    string owner = 2; 
+    string expiry_date = 3; 
+    name = 4; 
+    ProductList products = 5; 
+    repeated PropertyValue properties = 6; 
+} 
+```
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+The design of catalog in this manner leaves it extensible for domain specific
+use. The concept of price has been discussed and it's been decided to include a
+notion of price within products added to a catalog. A simple implmentation of
+price is fine and currently serves as a placeholder, while a more robust
+standard of modeling price data can be implemented from a future pricing-related
+RFC. A GS1 pricing standard that can be leveraged in the writing of such an RFC
+would be [GS1 Price
+Sync](https://www.gs1.org/docs/gdsn/3.1/BMS_Price_Sync_r3p1p3_i1p3p5_23May2017.pdf).
+
+# Prior art
+[prior-art]: #prior-art
+
+In the distributed ledger space there are not any example implementations of a
+product catalog. There are specs and standards around catalogs that can be
+leveraged in the design and implementation of a Grid Catalog.
+
+See Ariba's® [catalog interchange format
+(CIF)](https://www.essent.com/What-is-a-CIF-Catalog.html) CIF that has grown
+into a de facto standard and is widely used by hundreds of thousands of
+companies throughout the world.
+
+[GS1's Catalogue Item
+Synchronisation](https://www.gs1.org/docs/gdsn/3.1/BMS_GDSN_Catalogue_Item_Sync_r3p1p0_i1_p0_p6_25Aug2015.pdf)
+is the process of continuous harmonisation of item information between trading
+partners which ensures that the master data is the same in all trading partners
+systems.
+
```

Comment:
This is one spec, but is part of a much larger data synchronization network.  Can we get more information about the hundreds of thousands of companies using CIF and in what manner for what level of product? I went to the link but did not see users, or industry groups

Reply:
AFAIK it's something baked into SAP -- @davececchi may know more about it than I do.

I'll update the link, this has more details about CIF: https://public.cummins.com/sites/CSP/SiteCollectionDocuments/Help/AribaNetworkHelp/CIF_Catalog_Guide.pdf